code/LeftAndMain.php

```diff
@@ -1759,22 +1788,43 @@ public function setCurrentPageID($id)
      * to get the currently selected record.
      *
      * @return DataObject
+     * @deprecated 5.4.0 use currentRecord() instead.
      */
     public function currentPage()
     {
-        return $this->getRecord($this->currentPageID());
+        Deprecation::notice('5.4.0', 'use currentRecord() instead.');
+        return $this->currentRecord();
+    }
+
+    /**
+     * Uses {@link getRecord()} and {@link currentRecordID()}
+     * to get the currently selected record.
+     */
+    public function currentRecord(): ?DataObject
```

Comment:
This will be `null` before an edit page is opened, e.g. in `CMSMain` before selecting a page.